src/utils.jl

```diff
@@ -462,9 +462,12 @@ function flatten1(ex)
   return length(ex′.args) == 1 ? ex′.args[1] : ex′
 end
 
-# Helper - only flatten the arguments of x
-flatten_args(x::Expr) = Expr(x.head, map(flatten, x.args)...)
-flatten_args(x) = x
+# Helpers for flattening try blocks
+false_or_symbol_or_bflatten(x) = x==false || isa(x, Symbol) ? x : bflatten(x)
+function bflatten(x) # flatten down to a block (i.e. a single symbol is turned into `begin symbol end`, unlike `flatten` which would just return the symbol)
+    fx = flatten(x)
+    return isexpr(fx, :block) ? fx : Expr(:block,fx)
```

Comment:
If I `flatten(Expr(:try, :(sin(10)), false))`, then it is pretty odd that `flatten` will _add_ a new `:block`, which wasn't there before. That's the opposite of what you want from `flatten`. If this were a brand new function, I'd say that it's OK because that's what the Julia parser yields anyway, but given that we'll potentially break some code, I'd rather try to stick to the previous behaviour here:

```suggestion
    return isexpr(fx, :block) || !isexpr(x, :block) ? fx : Expr(:block,fx)
```

Reply:
Also, with this change, I don't think you need `false_or_symbol_or_bflatten`.

Reply:
I guess I did not want to be permissive, I wanted this to work only for `false` and `::Symbol`, because that is the only thing that Julia produces. But maybe someone would be making weird expressions, so I should not be that restrictive, as you mentioned above.